x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -508,22 +506,27 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     type: string | string[],
     options: SavedObjectsOpenPointInTimeOptions
   ) {
-    try {
-      const args = { type, options };
-      await this.legacyEnsureAuthorized(type, 'open_point_in_time', options?.namespace, {
+    const args = { type, options };
+    const { status, typeMap } = await this.legacyEnsureAuthorized(
+      type,
+      'open_point_in_time',
+      options?.namespaces,
+      {
         args,
         // Partial authorization is acceptable in this case because this method is only designed
         // to be used with `find`, which already allows for partial authorization.
         requireFullAuthorization: false,
-      });
-    } catch (error) {
+      }
+    );
+
+    if (status === 'unauthorized') {
       this.auditLogger.log(
         savedObjectEvent({
           action: SavedObjectAction.OPEN_POINT_IN_TIME,
-          error,
+          error: new Error(status),
         })
       );
-      throw error;
+      throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));
```

Comment:
@jportner I think I gonna need your help on this one

Changing the behavior by adding `throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));` here makes some of the security SO export tests fails (which is to be expected)

The failing tests are the ones regarding the `exportableTypes`, that were previously expecting a `200` code, and should now receive a `403`.

https://github.com/elastic/kibana/blob/5c3c3efdd87089fb1a326854c83397a7253bd7c6/x-pack/test/saved_object_api_integration/security_only/apis/export.ts#L54-L55

However, when changing the expected status code

```diff
-- createTestDefinitions(exportableTypes, { statusCode: 200, reason: 'unauthorized' })
++ createTestDefinitions(exportableTypes, { statusCode: 403, reason: 'unauthorized' })
```

The tests fails because the reason does not match

```
   Error: expected { statusCode: 403,
   error: 'Forbidden',
   message: 'unauthorized' } to sort of equal { statusCode: 403,
   error: 'Forbidden',
   message: 'Unable to bulk_get sharedtype' }
```

```diff
{
      "error": "Forbidden"
  -  "message": "unauthorized"
  +  "message": "Unable to bulk_get sharedtype"
      "statusCode": 403
}
```

and I just don't get where this message is coming from. 

Just to be sure I did not change anything else, I removed the throw statement in the security wrapper

```diff
-- throw SavedObjectsErrorHelpers.decorateForbiddenError(new Error(status));
++ return await this.baseClient.openPointInTimeForType(type, options)
```

and the tests correctly passes (with expected status code 200 I mean)

From a quick `Unable to ${` grep, the occurrences I found are:

https://github.com/elastic/kibana/blob/196531502f8cea18e484e05205fcb47990617300/x-pack/plugins/security/server/saved_objects/ensure_authorized.ts#L117-L118

https://github.com/elastic/kibana/blob/3fbfcad060baf3cd084bda4d9318afa9e5a3a334/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L897-L898

https://github.com/elastic/kibana/blob/3fbfcad060baf3cd084bda4d9318afa9e5a3a334/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L936-L937

However, I don't get how adding this `throw` statement can result on this specify error message. Do you have any insight?

Reply:
Hum, I read the diff wrong. What you specify to `createTestDefinition` is not the plain error message but an enum value to change the assertion behavior:

https://github.com/elastic/kibana/blob/59d7f331158d70919dd171b3b2684393d1f3d8e7/x-pack/test/saved_object_api_integration/common/suites/export.ts#L155-L159

Hum, we now have two kind of expected `403`.... not sure what the best way to handle this is?

Reply:
Hum, actually looking at the tests, the by-id export was already throwing a 403. Only the by-type export was returning an empty export (which explains the 403 vs 200 distinction in the test code), so both mode now returning 403 does make sense.